lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -475,26 +473,30 @@ def detach_eni(eni, module):
         module.exit_json(changed=False, interface=get_eni_info(eni))
 
 
-def find_eni(connection, module):
+def uniquely_find_eni(connection, module):
 
     eni_id = module.params.get("eni_id")
-    subnet_id = module.params.get('subnet_id')
     private_ip_address = module.params.get('private_ip_address')
     instance_id = module.params.get('instance_id')
     device_index = module.params.get('device_index')
 
     try:
         filters = {}
-        if subnet_id:
```

Comment:
Fixed in 28ba4c11a and tested with the following playbook snippet.

```
- name: Debug the ec2_eni module
  hosts: all
  vars:
    # NB: The two subnets have the same IP ranges (they can since they're in different VPCs)
    first_subnet_id: "<subnet-id-1>"
    second_subnet_id: "<subnet-id-2>"
  tasks:

    - name: Create a first ENI
      ec2_eni:
        subnet_id: "{{ first_subnet_id }}"
        private_ip_address: "10.0.1.10"
        description: "First ENI"
        state: "present"
      register: first_eni

    - name: Create a second ENI
      ec2_eni:
        subnet_id: "{{ second_subnet_id }}"
        description: "Second ENI"
        private_ip_address: "10.0.1.10"
        state: "present"

    - name: Modify the first ENI via eni_id
      ec2_eni:
        eni_id: "{{ first_eni['interface']['id'] }}"
        subnet_id: "{{ first_subnet_id }}"
        delete_on_termination: yes
        source_dest_check: no

    - name: Modify the second ENI via subnet_id and private_ip_address
      ec2_eni:
        subnet_id: "{{ second_subnet_id }}"
        private_ip_address: "10.0.1.10"
        description: "I should be Second ENI"
```